rust/src/ics23.rs

```diff
@@ -0,0 +1,128 @@
+extern crate failure;
+
+
+use crate::proofs;
+use crate::verify::{CommitmentRoot, verify_existence};
+
+pub fn verify_membership(proof: &proofs::CommitmentProof, spec: &proofs::ProofSpec, root: CommitmentRoot, key: &[u8], value: &[u8]) -> bool {
+    match &proof.proof {
+        Some(proofs::CommitmentProof_oneof_proof::exist(ex)) => {
+            let valid = verify_existence(&ex, spec, root, key, value);
+            valid.is_ok() && valid.unwrap()
+        }
+        _ => false,
+    }
```

Comment:
when matching on a single variant of an enum, you can avoid nesting by writing it as:
```
if let Some(proofs::CommitmentProof_oneof_proof::exist(ex))) = &proof.proof {
    let valid = verify_existence(&ex, spec, root, key, value);
    valid.is_ok() && valid.unwrap()
} else {
    false
}
```

Reply:
Ah... I only saw this after asking on zulip :man_facepalming: